src/linker/Linker/UnconditionalSuppressMessageAttributeState.cs

```diff
@@ -18,8 +18,18 @@ public class UnconditionalSuppressMessageAttributeState
 
 		public class Suppression
 		{
-			public SuppressMessageInfo SuppressMessageInfo { get; set; }
+			public SuppressMessageInfo SuppressMessageInfo { get; }
 			public bool Used { get; set; }
+			public CustomAttribute OriginAttribute { get; }
+			public ICustomAttributeProvider Provider { get; }
+
+			public Suppression (SuppressMessageInfo suppressMessageInfo, bool used, CustomAttribute originAttribute, ICustomAttributeProvider provider)
```

Comment:
The used is always set to false upon construction. You could remove the parameter from the .ctor and simply always initialize it to `false`.